docs/features/logforwarding/outputs/splunk-forwarding.adoc

```diff
@@ -43,5 +49,146 @@ spec:
+In such cases, users can configure the required fields manually.
+
+==== Indexed Fields Transformation Rules
+
+To support the use of complex and nested fields as indexed fields, field names and values are automatically transformed to meet requirements:
+
+* Nested fields are flattened into top-level fields.
+* Field paths are joined using dot notation, and unsupported characters are replaced with underscores (`_`).
+* Non-string values are automatically converted to strings (e.g., `3` → `"3"`, `true` → `"true"`).
+* Object values are serialized as JSON strings (e.g., `{ status: 200 }` → `"{\"status\":200}"`).
+
+These transformations allow for flexible field selection without requiring changes to the original log structure.
+
+=== Examples
+
+==== Example 1: Nested key
+
+ClusterLogForwarder configuration:
+[source,yaml]
+----
+indexedFields: [".annotations.authorization.k8s.io/decision", ".annotations.authorization.k8s.io/reason"]
+----
+
+Original log structure:
+[source,json]
+----
+"annotations": {
+  "authorization.k8s.io/decision": "allow",
+  "authorization.k8s.io/reason": "foo_bar"
+}
+----
+
+Transformed fields:
+[source,json]
+----
+"annotations_authorization_k8s_io_decision": "allow",
+"annotations_authorization_k8s_io_reason": "foo_bar"
+----
+
+==== Example 2: Numeric and Boolean Values
+
+ClusterLogForwarder configuration:
+[source,yaml]
+----
+indexedFields: [".status_code", ".success"]
+----
+
+Original log structure:
+[source,json]
+----
+"status_code": 200,
+"success": true
+----
+
+Transformed fields:
+[source,json]
+----
+"status_code": "200",
+"success": "true"
+----
+
+==== Example 3: Embedded Object
+
+ClusterLogForwarder configuration:
+[source,yaml]
+----
+indexedFields: [".objectRef.resource"]
+----
+
+Original log structure:
+[source,json]
+----
+"objectRef": {
+  "resource": {
+    "status": 200,
+    "statusText": "OK"
+  }
+}
+----
+
+Transformed field:
+[source,json]
+----
+"objectRef_resource": "{\"status\":200,\"statusText\":\"OK\"}"
+----
+
+[IMPORTANT]
+====
+After remapping, the *original fields* will be removed to avoid duplication.
+Only the transformed, top-level fields will be retained in the final log event.
+====
+
+=== `payloadKey`
+
+By default, `payloadKey` is not set, which means the complete log record is forwarded as the payload.
+
+
+NOTE: Use `payloadKey` carefully. Selecting a single field as the payload may cause other important information in the log to be dropped, potentially leading to inconsistent or incomplete log events.
+
+
+=== `sourceType` and `host`
+
+The `sourceType` and `host` fields are *not configurable* through the Cluster Log Forwarder API. They are set automatically as described below, independent of the `.log_type` or `.log_source` values.
+
+* `sourceType`: Set to `_json`
```

Comment:
This statement will need to change depending upon the evaluation of what is sent to event:

* event == object:  _json
* event == string:  _text (??)
* event == array:   _json(??)
* event == numeric : _text(?? do we convert to a string)